sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -684,6 +719,7 @@ Scheduler::GraphBuilder::addCG(std::unique_ptr<detail::CG> CommandGroup,
   // Set new command as user for dependencies and update leaves.
   // Node dependencies can be modified further when adding the node to leaves,
   // iterate over their copy.
+  // FIXME employ a reference here to eliminate copying of a vector
```

Comment:
You could just iterate over NewCmd->MDeps directly without copying it or creating a reference to it.

Reply:
Sure thing. This `FIXME` is a mere note on place to improve somehow.